extensions/import/src/dialogs/derivedColumnDialog.ts

```diff
@@ -0,0 +1,376 @@
+					{ value: c.columnName }
+				);
+				columnTableData.push(tableRow);
+			});
+			this._view = view;
+			const columnTable = view.modelBuilder.declarativeTable().withProps({
+				columns: [
+					{
+						displayName: '',
+						valueType: azdata.DeclarativeDataType.boolean,
+						isReadOnly: false,
+						showCheckAll: true,
+						width: '20px',
+						headerCssStyles: headerLeft,
+						rowCssStyles: styleLeft
+					},
+					{
+						displayName: constants.columnTableTitle,
+						valueType: azdata.DeclarativeDataType.string,
+						isReadOnly: true,
+						width: '140px',
+						headerCssStyles: headerLeft,
+						rowCssStyles: styleLeft
+					}
+				],
+				dataValues: columnTableData,
+				CSSStyles: {
+					'table-layout': 'fixed'
+				}
+			}).component();
+
+			console.log(columnTableData);
+
+			columnTable.onDataChanged(e => {
+				if (e.value) {
+					this._transformationTable.columns.splice(this._transformationTable.columns.length - 1, 0,
+						{
+							displayName: this._model.proseColumns[e.row].columnName,
+							valueType: azdata.DeclarativeDataType.string,
+							isReadOnly: true,
+							width: '100px',
+							headerCssStyles: headerLeft,
+							rowCssStyles: styleLeft
+						}
+					);
+					for (let index = 0; index < this._model.proseDataPreview.length; index++) {
+						this._transformationTable.dataValues[index].splice(
+							this._transformationTable.dataValues[index].length - 1,
+							0,
+							{ value: this._model.proseDataPreview[index][e.row] }
+						);
+					}
+
+				}
+				else {
+					let removeIndex = 0;
+					for (let index = 0; index < this._transformationTable.columns.length; index++) {
+						if (this._model.proseColumns[e.row].columnName === this._transformationTable.columns[index].displayName) {
+							removeIndex = index;
+							break;
+						}
+					}
+					this._transformationTable.columns.splice(removeIndex, 1);
+					for (let index = 0; index < this._model.proseDataPreview.length; index++) {
+						this._transformationTable.dataValues[index].splice(removeIndex, 1);
+					}
+				}
+				const isColumnAdded = this._transformationTable.columns.length > 1;
+				this.clearAndAddTransformationContainerComponents(isColumnAdded);
+				this._applyButton.enabled = isColumnAdded;
+			});
+
+
+			const columnContainer = view.modelBuilder.flexContainer().withLayout({
+				flexFlow: 'column',
+				width: '180px',
+				height: '100%'
+			}).withProps({
+				CSSStyles: {
+					'border-right': 'solid 1px'
+				}
+			}).component();
+			columnContainer.addItem(columnTable);
+
+			const transformationTableData: azdata.DeclarativeTableCellValue[][] = [];
+			for (let index = 0; index < this._model.proseDataPreview.length; index++) {
+				const tableRow: azdata.DeclarativeTableCellValue[] = [];
+				this._specifyTransformations.push(this._view.modelBuilder.inputBox().withProps({
+					value: '',
+					placeHolder: constants.specifyTransformation
+				}).component());
+				tableRow.push({
```

Comment:
Could also just do 

`transformationTableData.push([{ value: this._specifyTransformations[index] }]);`